service/javadsl/client/src/main/java/com/lightbend/lagom/javadsl/client/ConfigurationServiceLocator.java

```diff
@@ -41,34 +42,51 @@ public ConfigurationServiceLocator(Configuration configuration, CircuitBreakers
         this(configuration.underlying(), new CircuitBreakersPanelImpl(circuitBreakers));
     }
 
-    @Inject
-    public ConfigurationServiceLocator(Config config, CircuitBreakersPanel circuitBreakersPanel) {
-        super(circuitBreakersPanel);
+  @Inject
+  public ConfigurationServiceLocator(Config config, CircuitBreakersPanel circuitBreakersPanel) {
+      super(circuitBreakersPanel);
 
-        Map<String, URI> services = new HashMap<>();
+      Map<String, List<URI>> services = new HashMap<>();
 
-        if (config.hasPath(LAGOM_SERVICES_KEY)) {
-            Config configServices = config.getConfig(LAGOM_SERVICES_KEY);
-            for (String key: configServices.root().keySet()) {
-                try {
-                    String value = configServices.getString(key);
-                    URI uri = new URI(value);
-                    services.put(key, uri);
-                } catch (ConfigException.WrongType e) {
-                    throw new IllegalStateException("Error loading configuration for " + getClass().getSimpleName() + ". Expected lagom.services." + key + " to be a String, but was " + configServices.getValue(key).valueType(), e);
-                } catch (URISyntaxException e) {
-                    throw new IllegalStateException("Error loading configuration for  " + getClass().getSimpleName() + ". Expected lagom.services." + key + " to be a URI, but it failed to parse", e);
-                }
-            }
-        }
-        this.services = HashTreePMap.from(services);
-    }
+      if (config.hasPath(LAGOM_SERVICES_KEY)) {
+          Config configServices = config.getConfig(LAGOM_SERVICES_KEY);
 
+          for (String key: configServices.root().keySet()) {
+            List<String> endpoints = ConfigExtensions.getStringList(configServices, key);
+            List<URI> uris = endpoints
+                    .stream()
+                    .map(uri -> {
+                      try {
+                        return new URI(uri);
+                      } catch (ConfigException.WrongType e) {
```

Comment:
Another option might be to use [`URI.create`](https://docs.oracle.com/javase/8/docs/api/java/net/URI.html#create-java.lang.String-) (which throws a runtime exception) instead of `new URI`. But it's fine to split it into two separate try/catch blocks, too.